### ISSUE TITLE
Safari/Webkit WebRTC audio delay

### ISSUE THREAD

**Issue by bchah** [AirenSoft/OvenPlayer#179]
WebRTC playback with OvenPlayer in Safari is always out of sync, where the audio is behind the video.

Any recent version of Safari (14+) on macOS or iOS exhibits the issue. But accessing the same stream in Chromium / FireFox is 100% flawless (Amazing work on the OvenMediaEngine and player, guys! 😀)

I have reproduced this with a number of connections, hardware and software encoders, and client environments. OvenMediaEngine instances have been running on Docker with macOS or Linux hosts - I started testing with v0.10.8 but the issue is the same on all new versions. Same with OvenPlayer, trying many builds and it is the same. There are no playback errors in the server log but here is the incoming stream info, which looks pretty good:

```
[Stream Info]
id(135255429), output(e1d4fa2016f8), SourceType(Transcoder), Created Time (Fri Feb 12 19:45:57 2021)
	>> Origin Stream Info
	id(24), output(e1d4fa2016f8), SourceType(Rtmp), Created Time (Fri Feb 12 19:45:57 2021)

	Video Track #0: Bypass(true) Bitrate(4.00Mb) codec(1, H264) resolution(1600x900) framerate(60.00fps) timebase(1/1000)
	Audio Track #1: Bypass(false) Bitrate(128.00Kb) codec(8, OPUS) samplerate(48.0K) format(s16, 16) channel(stereo, 2) timebase(1/48000)
```

Are you able to reproduce this issue with Safari? Do you have any idea what could be causing it?

**Comment by SangwonOh** [comment]
@bchah Hi. It seems there's an audio delay issue. Could you check that is there a same issue on older version of Safari (< 14) ?

**Comment by bchah** [comment]
Hi @SangwonOh thank you for the response 🥇

I tested with Safari 12.1.2 (macOS Mojave 10.14.6) and you are right, the playback is perfect! Safari 13 is hard to find for testing because any device which received Safari 13 also received Safari 14, and Apple is not friendly with downgrades.

So there must have been a regression in either Safari 13 or 14 which affected WebRTC.

Is there anything more I can do to help you solve this? Thank you again for looking at the issue.

**Comment by SangwonOh** [comment]
@bchah Hi. Thank you for you're testing. Could you give us the Server.xml file of OvenMediaEngine ? I will analyze this problem with the OvenMediaEngine team.

**Comment by bchah** [comment]
Hi @SangwonOh thank you for the followup. Attached is a sample Server XML and some stream configurations too, but I have tried a few different configurations too. The latency gets a bit better if you transcode to VP8 instead of using a passthrough video stream, but is still behind in Safari 14 versus perfect in Safari 12, Chrome or Firefox.

For input streams I have tried your recommended settings in OBS (CBR, 1s GOP, baseline profile, zerolatency tuning and bframes=0 flag). I also have a few types of hardware encoder boards (like the V310 if you are familiar?) and I am having much success as long as the profile is baseline and GOP is 1s.

[server-sample.xml.zip](https://github.com/AirenSoft/OvenPlayer/files/5983585/server-sample.xml.zip)
![hardware-encoder](https://user-images.githubusercontent.com/33604674/107985755-b1955c80-6f98-11eb-99a6-2d92a31cfcdd.jpg)
![obs](https://user-images.githubusercontent.com/33604674/107985770-b9550100-6f98-11eb-9fa3-3e73391d1a47.jpg)
